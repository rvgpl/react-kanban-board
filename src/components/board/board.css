.board {
  max-width: 1200px;
  margin: 50px auto;
}
.board-input-field {
  display: flex;
  align-items: center;
  justify-content: space-between;
  gap: 1rem;
  margin-bottom: 2rem;
}

.board-lanes {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr;
  gap: 1.4rem;
}