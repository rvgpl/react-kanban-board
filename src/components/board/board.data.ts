export const todoTasks = [
  {
    id: 1,
    text: "Improve website UI polish",
    createdAt: 1646412499232,
  },
  {
    id: 2,
    text: "Disable button on page 4",
    createdAt: 1646412499232,
  },
  {
    id: 3,
    text: "Refactor card component",
    createdAt: 1646412499232,
  },
];

export const inProgressTasks = [
  {
    id: 4,
    text: "Create a Kanban board",
    createdAt: 1646412499232,
  },
];

export const doneTasks = [
  {
    id: 4,
    text: "Create a sample react project",
    createdAt: 1646412499232,
  },
];
