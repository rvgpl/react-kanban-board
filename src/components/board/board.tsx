import React, { useState } from "react";
import { Button } from "../button/button";
import { Card } from "../card/card";
import { Lane } from "../lane/lane";
import { TextField } from "../text-field/text-field";

import "./board.css";
import { doneTasks, inProgressTasks, todoTasks } from "./board.data";

const Board: React.FC = () => {
  const [value, setValue] = useState("");
  return (
    <section className="board">
      <div className="board-input-field">
        <TextField value={value} onChange={setValue} />
        <Button onClick={() => console.log(value)}>Add</Button>
      </div>
      <div className="board-lanes">
        <Lane title="Todo">
          {todoTasks.map((task) => (
            <Card
              task={task}
              key={task.id}
              onClick={(id) => console.log("card id", id)}
            />
          ))}
        </Lane>
        <Lane title="In Progress">
          {inProgressTasks.map((task) => (
            <Card
              task={task}
              key={task.id}
              onClick={(id) => console.log("card id", id)}
            />
          ))}
        </Lane>
        <Lane title="Done">
          {doneTasks.map((task) => (
            <Card
              task={task}
              key={task.id}
              onClick={(id) => console.log("card id", id)}
              disableButton
            />
          ))}
        </Lane>
      </div>
    </section>
  );
};

export { Board };
