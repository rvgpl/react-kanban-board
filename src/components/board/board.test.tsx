import { render } from "@testing-library/react";
import { Board } from "./board";

describe("Board", () => {
  it("should render", () => {
    const { container } = render(<Board />);
    expect(container).toMatchSnapshot();
  });
});
