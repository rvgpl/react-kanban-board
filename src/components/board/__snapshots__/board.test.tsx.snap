// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board should render 1`] = `
<div>
  <section
    class="board"
  >
    <div
      class="board-input-field"
    >
      <input
        class="text-field"
        data-testid="text-field"
        value=""
      />
      <button
        class="button"
        data-testid="button"
      >
        Add
      </button>
    </div>
    <div
      class="board-lanes"
    >
      <section
        class="lane"
      >
        <h2
          class="lane-title"
        >
          Todo
        </h2>
        <div
          class="lane-items"
        >
          <div
            class="card"
          >
            <p
              class="card-text"
            >
              Improve website UI polish
            </p>
            <div
              class="card-footer"
            >
              <time
                class="card-timestamp"
              >
                4-3-2022 20:48
              </time>
              <button
                class="button"
                data-testid="button"
              >
                Move &gt;
              </button>
            </div>
          </div>
          <div
            class="card"
          >
            <p
              class="card-text"
            >
              Disable button on page 4
            </p>
            <div
              class="card-footer"
            >
              <time
                class="card-timestamp"
              >
                4-3-2022 20:48
              </time>
              <button
                class="button"
                data-testid="button"
              >
                Move &gt;
              </button>
            </div>
          </div>
          <div
            class="card"
          >
            <p
              class="card-text"
            >
              Refactor card component
            </p>
            <div
              class="card-footer"
            >
              <time
                class="card-timestamp"
              >
                4-3-2022 20:48
              </time>
              <button
                class="button"
                data-testid="button"
              >
                Move &gt;
              </button>
            </div>
          </div>
        </div>
      </section>
      <section
        class="lane"
      >
        <h2
          class="lane-title"
        >
          In Progress
        </h2>
        <div
          class="lane-items"
        >
          <div
            class="card"
          >
            <p
              class="card-text"
            >
              Create a Kanban board
            </p>
            <div
              class="card-footer"
            >
              <time
                class="card-timestamp"
              >
                4-3-2022 20:48
              </time>
              <button
                class="button"
                data-testid="button"
              >
                Move &gt;
              </button>
            </div>
          </div>
        </div>
      </section>
      <section
        class="lane"
      >
        <h2
          class="lane-title"
        >
          Done
        </h2>
        <div
          class="lane-items"
        >
          <div
            class="card"
          >
            <p
              class="card-text"
            >
              Create a sample react project
            </p>
            <div
              class="card-footer"
            >
              <time
                class="card-timestamp"
              >
                4-3-2022 20:48
              </time>
            </div>
          </div>
        </div>
      </section>
    </div>
  </section>
</div>
`;
